src/Resources/skeleton/resetPassword/ResetPasswordController.tpl.php

```diff
@@ -75,13 +76,18 @@ public function request(Request $request, MailerInterface $mailer): Response
 <?php } ?>
     public function checkEmail(): Response
     {
+        $generatedAt = $this->getTokenGeneratedAtFromSession();
+
         // We prevent users from directly accessing this page
-        if (!$this->canCheckEmail()) {
+        if (null === $generatedAt || !$this->canCheckEmail()) {
             return $this->redirectToRoute('app_forgot_password_request');
         }
 
+        // We need to generate a dummy token to retrieve the translation data for the template.
+        $dummyToken = new ResetPasswordToken('fake', new \DateTimeImmutable(), $generatedAt);
```

Comment:
This is kind of a bummer (which I'm sure you already know!).

Could we just store the entire `ResetPasswordToken` in the session? That would replace the current boolean we use in the `canCheckEmail()` check and the new date. 

If we're worried about the token string inside `ResetPasswordToken` being stored in the session, we could call a method to clear the token string from the `ResetPasswordToken` object before storing it in the session (and then, throw an exception if the user tries to access it). A new controller trait could handle cloning the object, clearing the token, and setting it on the session.

WDYT? It's a bit odd... but then we're basically keeping the `ResetPasswordToken` object in the session instead of dates or booleans.

Reply:
I like it.. I was thinking about that earlier only being concerned by the token (clearing it as suggested would solve that problem) and performance / tangible costs (e.g. AWS Redis for larger applications). I doubt the latter would ever become a factor in the majority of most applications, but if it did, I'm sure the dev team could implement a more cost effective solution. 

I'll refactor to go this route - v2 of reset bundle may provide us with a better solution in the future. None of which are feasible now without BC.

Reply:
done